t14_extended_source/runAnalysis.py

```diff
@@ -0,0 +1,254 @@
+  return np.array(edep_annihilation_recs, dtype=np.float32), np.array(edep_prompt_recs, dtype=np.float32), expected_entries
+
+def perform_ks_test(edeps, gamma_type, source_type, decay_type):
+  """
+  Performs the two-sample Kolmogorov-Smirnov test for given data from simulation and reference data (data/ directory).
+  Null hypothesis: two distributions are identical.
+
+  Parameters
+  ----------
+  edep : np.array
+   energy depositions
+  gamma_type : GammaType
+  source_type : SourceType
+  decayType : DecayType
+
+  Returns
+  -------
+  passed test : bool
+  """
+  source_strings = {SourceType.PARAPOSITRONIUM: "pPs", SourceType.ORTHOPOSITRONIUM: "oPs"}
+  decay_strings = {DecayType.STANDARD:"",DecayType.WITHDEEXCITATION:"Prompt"}
+  gamma_type_strings = {GammaType.ANNIHILATIONGAMMA:"-annihilation-gamma",GammaType.PROMPTGAMMA:"-prompt-gamma"}
+  script_dir_path = Path(os.path.dirname(os.path.abspath(__file__)))
+  tests_data_dir_path = script_dir_path/"data"
+  file_path = tests_data_dir_path/"".join([source_strings[source_type],decay_strings[decay_type],gamma_type_strings[gamma_type],".csv"])
+  with file_path.open(mode="r") as file:
+    csvr = csv.reader(file)
+    next(csvr)
+    test_edeps = list()
+    for edep in csvr:
+      test_edeps.append(float(edep[0]))
+    test_edeps = np.array(test_edeps, dtype=np.float32)
+  _, pvalue = stats.ks_2samp(edeps, test_edeps)
+  #in scipy.stats.ks_2samp a hypothesis is for two-sided test is: two distributions are identical, F(x)=G(x) for all x
+  #We can say with certainty that our hypothesis is not true for pvalue equals 5% or lower.
+  pvalue_threshold = 0.05
+  return pvalue > pvalue_threshold
+
+
+def analyse_model(tree, edep_plots, source_type, decay_type):
+  """
+  Checks if data in tree are correct:
+  * number of entries from prompt and annihilation gammas equals entries from the model
+  * energy depositions records passed Kolmogorov–Smirnov test
+
+  Parameters
+  ----------
+  tree : uproot tree
+  edep_plots : dict
+   for given model new records are added by this method
+  source_type : sourceType
+  decay_type : decayType
+
+  Returns
+  -------
+  correct data (tests passed) : bool
+  """
+  source_strings = {SourceType.PARAPOSITRONIUM: "pPs", SourceType.ORTHOPOSITRONIUM: "oPs"}
+  if not source_type in source_strings:
+    return False
+  if not decay_type in {DecayType.STANDARD, DecayType.WITHDEEXCITATION}:
+    return False
+
+  edep_annihilation, edep_prompt, expected_entries = get_edep_plot_data(tree, source_type, decay_type)
+
+  total_edep_len = len(edep_annihilation) + len(edep_prompt)
+  if expected_entries != total_edep_len:
+    return False
+
+  if decay_type == DecayType.STANDARD:
+    edep_plots[source_strings[source_type]] = edep_annihilation
+    return perform_ks_test(edep_annihilation, GammaType.ANNIHILATIONGAMMA, source_type, decay_type)
+  plot_title = "".join([source_strings[source_type],"Prompt - annihilation gamma"])
```

Comment:
Do I understand correctly taht if DecayType is not STANDARD then the plot is created?

Reply:
In both cases is created - in first case (STANDARD) as a title is used source_strings[source_type], but for the second type (DecayType.WITHDEEXCITATION) we have to separate edep plots for annihilation and prompt gamma - so we have to set different titles for edep plots.